src/relayclient/GsnTestEnvironment.ts

```diff
@@ -0,0 +1,148 @@
+import net from 'net'
+import { ether } from '@openzeppelin/test-helpers'
+
+import CommandsLogic, { DeploymentResult } from '../cli/CommandsLogic'
+import KeyManager from '../relayserver/KeyManager'
+
+import { configureGSN } from './GSNConfigurator'
+import { getNetworkUrl } from '../cli/utils'
+import { TxStoreManager } from '../relayserver/TxStoreManager'
+import RelayServer from '../relayserver/RelayServer'
+import HttpServer from '../relayserver/HttpServer'
+import { Address } from './types/Aliases'
+import { RelayProvider } from './RelayProvider'
+import Web3 from 'web3'
+
+export interface TestEnvironment {
+  deploymentResult: DeploymentResult
+  relayProvider: RelayProvider
+  httpServer: HttpServer
+  relayUrl: string
+}
+
+class GsnTestEnvironmentClass {
+  private httpServer?: HttpServer
+
+  /**
+   *
+   * @param host:
+   * @param paymaster TODO: will allow using custom paymaster (need to provide ABI file contents)
+   * @return
+   */
+  async startGsn (host: string, paymaster?: any): Promise<TestEnvironment> {
+    await this.stopGsn()
+    const _host: string = getNetworkUrl(host)
+    if (_host == null) {
+      throw new Error(`Failed to connect to ${host}`)
+    }
+    const commandsLogic = new CommandsLogic(_host, configureGSN({}))
+    const from = await commandsLogic.findWealthyAccount()
+    if (from == null) {
+      throw new Error('could not get unlocked account with sufficient balance')
+    }
+    const deploymentResult = await commandsLogic.deployRelayHub(from, undefined, paymaster)
```

Comment:
why not keep `deploymentInfo` as a member? 
we can `save()` it, and then later `load()` and use the same contracts

Reply:
Added to backlog